pkg/chains/signing.go

```diff
@@ -57,22 +78,58 @@ type TaskRunSigner struct {
 	SecretPath        string
 }
 
-// IsSigned determines whether a TaskRun has already been signed.
-func IsSigned(tr *v1beta1.TaskRun) bool {
+// SignedState returns TaskRun state.
+func SignedState(tr *v1beta1.TaskRun) State {
 	signature, ok := tr.ObjectMeta.Annotations[ChainsAnnotation]
 	if !ok {
-		return false
+		return Unknown
+	}
+
+	if value, ok := signedStates[signature]; !ok {
+		return Unknown
+	} else {
+		return value
 	}
-	return signature == "true"
+}
+
+// returns true if:
+// the signed annotation isn't there, the failed annotation isn't there, and retries<3
+func RetryAgain(tr *v1beta1.TaskRun) bool {
+	_, ok := tr.ObjectMeta.Annotations[ChainsRetriesAnnotation]
```

Comment:
this function isn't doing anything except for checking if ChainsRetriesAnnotation exists -- we want it to return true if ChainsRetriesAnnotation<3 OR if ChainsRetriesAnnotation doesn't exist yet